### PR TITLE
SIP328 fix wood storage bookkeeping so handling of `plantWoodCStorageDelta` is consistent

### LINKED ISSUE #328: Clarify SIPNET wood-storage bookkeeping around harvest and outputs
## Summary

SIPNET tracks wood carbon in two pieces:

```text
structural wood = envi.plantWoodC
storage delta   = envi.plantWoodCStorageDelta
reported wood   = structural wood + storage delta
```

storage delta can be negative because it is a bookkeeping. But SIPNET does not clearly define or test what should happen when this storage term interacts with harvest, turnover, respiration, and output reporting.

Related: 
- #325 
- #324

### Harvest calculates affected wood using:

```c
woodC = envi.plantWoodC + envi.plantWoodCStorageDelta;
```

but the event update applies the wood change only to:

```c
envi.plantWoodC
```

and does not update `envi.plantWoodCStorageDelta`.

That means a full aboveground harvest can remove more wood than exists in the structural pool, after which structural wood is clamped to zero while positive storage remains. The result is that `sipnet.out` can still report positive `plantWoodC` after a harvest that appears intended to remove all aboveground wood.

### Negative storage is allowed, but if:

```text
envi.plantWoodC + envi.plantWoodCStorageDelta < 0
```

then wood respiration and wood turnover can become negative because they use this sum directly.

## Suggested fix

Make the intended functionality explicit and add focused tests:

1. Decide whether harvest acts on structural wood, reported/effective wood, or both.
2. Prevent negative effective wood from creating negative respiration or turnover fluxes.
3. Update the output docs for `plantWoodC` and `nppStorage`.
4. Add tests for harvest with positive and negative storage.

### PR DESCRIPTION
**Pull request by dlebauer**
## Summary

- **What**: Fixes wood-storage bookkeeping so harvest, wood respiration, and wood turnover handle `plantWoodCStorageDelta` consistently.
- **Motivation**: 
    - Make wood carbon storage bookkeeping internally consistent, satisfying mass-balance. 
    - Harvest, respiration, turnover, and output reporting should handle structural wood and storage delta with explicit semantics so SIPNET does not produce misleading reported wood stocks or nonphysical wood fluxes.
    - The current storage behavior can leave reported wood after harvest or create nonphysical negative wood fluxes. 
    
Without this fix, I was unable to find plausible parameterization that produced sensible AGB values following double counting of woodstorage that was fixed in #248.

## How was this change tested?

```bash
make libs/libsipnet.a
make -C tests/sipnet/test_modeling tests run
```

Added focused tests for:
- `sipnet.out` reporting `plantWoodC = structural wood + storage delta`
- `nppStorage = plantWoodCStorageDelta`
- negative storage not reducing wood respiration or wood turnover below structural wood
- harvest removing positive wood storage
- harvest leaving negative storage debt visible while preserving carbon balance

Still useful follow-up coverage:
- event-type tests that exercise harvest storage routing through the public event test fixtures
- restart tests with positive and negative `plantWoodCStorageDelta`
- smoke/reference-output review if this changes expected outputs in configured scenarios

## Still TODO

- [ ] Update Docs
- [ ] Update related issues 324,325 

## Reproduction steps

```bash
make libs/libsipnet.a
make -C tests/sipnet/test_modeling tests run
```

## Related issues

- Fixes #328
    - and maybe #324
- Related: #325, #324 #208, #248 

For #324, this fixes 
- Harvest no longer uses a negative plantWoodC + plantWoodCStorageDelta value to create negative litter/event fluxes.
- Wood respiration and wood turnover no longer become negative when storage debt makes reported total wood negative.
- Positive storage is explicitly routed during harvest, so full harvest removes positive storage instead of leaving reported wood behind.
- It **does not** implement a joint clamp/floor on plantWoodC + plantWoodCStorageDelta, which could invent carbon to erase carbon debt.

For #325, this ensures that negative wood storage does not make wood-based process fluxes negative.

## Checklist

- [x] Related issues are listed above.
- [ ] PR title has the issue number in it ("[#<number>] <concise description of proposed change>")
- [x] Tests added/updated for new features (if applicable)
- [ ] Documentation updated (if applicable)
  - Remaining docs work: clarify in `docs/user-guide/model-outputs.md` that text `plantWoodC` is reported total wood (`plantWoodC + nppStorage`) and structural wood is `plantWoodC - nppStorage`.
- [ ] `docs/CHANGELOG.md` updated with noteworthy changes
- [x] Code formatted with `clang-format` (run `git clang-format` if needed)

### CODE REVIEW

File: src/sipnet/state.c [hunk 1]
```diff
@@ -12,3 +14,11 @@ Envi envi;  // state variables
 Trackers trackers;
 PhenologyTrackers phenologyTrackers;
 Fluxes fluxes;
+
+double getPlantWoodCTotal(void) {
+  return envi.plantWoodC + envi.plantWoodCStorageDelta;
+}
+
+double getStorageBackedWoodCarbon(void) {
```
Comment (infotroph) [inline src/sipnet/state.c:22]: Ca you explain the name here? I'm not following what "storage backed" is supposed to mean.

**Comment by dlebauer** [comment]
closing as non-mass conserving. This is a band aid, and a more defensible fix will be developed in #325
